### ISSUE TITLE
Don't wrap captured header values in a vector

### ISSUE THREAD

**Issue by danielcompton** [steffan-westcott/clj-otel#9]
On this line, header values are wrapped in a vector:

https://github.com/steffan-westcott/clj-otel/blob/ba3ea1e36c3fe7c7b0d6ec87b2f5e2aae2494c1c/clj-otel-api/src/steffan_westcott/clj_otel/api/trace/http.clj#L23

What do you think about just making this a single value without a wrapping vector? It doesn't seem to be adding much here because only a single value is ever put in the vector.

**Comment by steffan-westcott** [comment]
It is worth remembering that HTTP headers may be assigned multiple values in any request or response message, they are not necessarily single-valued.

The [OpenTelemetry semantic conventions for HTTP request and response headers](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/http/#http-request-and-response-headers) states that the value must be an array (see note [2]). That array may be of several strings or a single string. In the case of the array of a single string, that string may have commas delimiting multiple values for the same header.

The [Ring spec for request `headers` map](https://github.com/ring-clojure/ring/blob/94d47951d9a70a4aed79fbbcc5631104d8af7b03/SPEC#L98) states that a `headers` value is a string, with commas delimiting multiple values for the same header.

To follow the OpenTelemetry semantic conventions, `clj-otel` places the single string into a vector, as you have noted. The string value provided by Ring is in the correct format expected by the semantic conventions.

In summary, I believe the current implementation in `clj-otel` follows the OpenTelemetry semantic conventions and should not be changed. Do you have a counterexample that shows this is not the case?

**Comment by danielcompton** [comment]
> In summary, I believe the current implementation in clj-otel follows the OpenTelemetry semantic conventions and should not be changed. Do you have a counterexample that shows this is not the case?

Nope, I think you're right. It isn't quite as nicely presented in Honeycomb as a bare value would be, but I can live with that :)
